### PR TITLE
RSDK-2498: adding intel up4000

### PR DESCRIPTION
**Pull request by susmitaSanyal**
Adding new board component; Intel up4000. 
Tested genericlinux changes on pi. 

TODO: 

- [ ] Test on jetson and beagle

- [ ] Test HW PWM

### CODE REVIEW

File: components/board/intel/board.go
Comment (randhid) [inline components/board/intel/board.go:3]: [minor] add tested models, datasheets or references, etc.

File: components/board/intel/board.go
Comment (randhid) [inline components/board/intel/board.go:12]: [major] We don't have a scop with a model name yet, so let's not register this fully just yet, as that is a breaking change.

File: components/board/genericlinux/data.go
Comment (susmitaSanyal) [reply components/board/genericlinux/data.go:106]: Agreed. Does RDK store architecture information? if not I can create a function.

**Comment by penguinland** [comment]
> Tested genericlinux changes on pi.

Rand is right: the pi is the only board that _doesn't_ use genericlinux, and we should test this on a BeagleBone and/or Jetson.

File: components/board/genericlinux/data.go
Comment (penguinland) [inline components/board/genericlinux/data.go:112]: I guess this works, but the helper function I would have created is `getCompatibles`. Something like this:

```
func getCompatibles(modelName string) []string, err {
    const (
        compatiblePathDt = "/proc/device-tree/compatible"
        compatiblePathDmi = "/sys/devices/virtual/dmi/id/board_name"
    )
    if compatiblePathDmi is a valid file:
        return a length-1 StringSet containing its contents

    read compatiblePathDt, return its contents split on null bytes
}
```

Making the `path` variable have two different purposes and passing that around seems like extra mental overhead for anyone working on this code, and then trying to write a single implementation of `newStringSetFromFile` that supports both device-tree and DMI boards seems hard.

File: components/board/genericlinux/data.go
Comment (penguinland) [inline components/board/genericlinux/data.go:102]: [minor] This sounds like it's for all boards, but you've implemented the rest as though this is only for device-tree boards. I'd put a `Dt` suffix on the name, similar to the `Dmi` suffix on the next line.

File: components/board/intel/board.go
Comment (penguinland) [reply components/board/intel/board.go:12]: Agreed: there are likely to be many Intel boards, but this only supports one of them for now. Maybe this should be "up4000", or maybe the plan is to expand this to support lots of boards by having lots of definitions in data.go? but I'd like a plan for how this works before we commit to such a broad model name.

**Review by penguinland — CHANGES_REQUESTED** [review]
The proof that this works should be successfully controlling GPIO pins on the Jetson and Up4000. Doing anything on the rpi is unrelated.

File: components/board/genericlinux/data.go
Comment (randhid) [reply components/board/genericlinux/data.go:106]: https://pkg.go.dev/runtime#pkg-constants

File: components/board/intel/board.go
Comment (susmitaSanyal) [reply components/board/intel/board.go:12]: Will update this when we decide on the name in scope doc. Till then, I'll keep "intel" as a placeholder.

File: components/board/register/register.go
Comment (susmitaSanyal) [reply components/board/register/register.go:10]: - [ ] remove component from register.

File: components/board/genericlinux/data.go
Comment (susmitaSanyal) [reply components/board/genericlinux/data.go:129]: a lot of these extra line was added and removed while I was printing out debug statements. Please ignore those, I will address them as they show up.

File: components/board/register/register.go
Comment (randhid) [reply components/board/register/register.go:10]: hold off on this for now - I'm going to see if we can get a name today to unblock you.

Comment (randhid) [reply components/board/register/register.go:10]:
```suggestion
	_ "go.viam.com/rdk/components/board/upboard"
```

File: components/board/intel/board.go
Comment (randhid) [inline components/board/intel/board.go]:
```suggestion
const modelName = "upboard"
```

File: components/board/intel/board.go
Comment (randhid) [inline components/board/intel/board.go]:
```suggestion
// Package upboard implements an Intel based board.
```

File: components/board/intel/board.go
Comment (randhid) [inline components/board/intel/board.go]:
```suggestion
package upboard
```

File: components/board/upboard/data.go
Comment (randhid) [inline components/board/upboard/data.go:5]:
```suggestion
const upboard = "up4000"
```

**Pull request closed without merging**
